src/pysdccc/_runner.py

```diff
@@ -0,0 +1,333 @@
+            raise ValueError(msg)
+        if not requirements.is_absolute():
+            msg = 'Path to requirements file must be absolute'
+            raise ValueError(msg)
+        if list(self.test_run_dir.iterdir()):
+            msg = f'{self.test_run_dir} is not empty'
+            raise ValueError(msg)
+
+        kwargs['no_subdirectories'] = 'true'
+        kwargs['test_run_directory'] = self.test_run_dir
+        kwargs['config'] = config
+        kwargs['testconfig'] = requirements
+        return _common.build_command(*args, **kwargs)
+
+
+class SdcccRunner(_BaseRunner):
+    """Synchronous runner for sdccc."""
+
+    def run(
+        self,
+        *,
+        config: _common.PATH_TYPE,
+        requirements: _common.PATH_TYPE,
+        timeout: float | None = None,
+        **kwargs: _common.CMD_TYPE,
+    ) -> tuple[int, TestSuite | None, TestSuite | None]:
+        """Run the SDCcc executable using the specified configuration and requirements.
+
+        This method executes the SDCcc executable with the provided configuration and requirements files,
+        and additional command line arguments. It logs the stdout and stderr of the process and waits for the
+        process to complete or timeout.
+        Checkout more parameter under https://github.com/draegerwerk/sdccc?tab=readme-ov-file#running-sdccc
+
+        :param config: The path to the configuration file. Must be an absolute path.
+        :param requirements: The path to the requirements file. Must be an absolute path.
+        :param timeout: The timeout in seconds for the SDCcc process. If None, wait indefinitely.
+        :param kwargs: Additional command line arguments to be passed to the SDCcc executable.
+        :return: A tuple containing the returncode of the sdccc process, parsed direct and invariant test results as
+        TestSuite objects.
+        :raises ValueError: If the provided paths are not absolute.
+        :raises subprocess.TimeoutExpired: If the process is running longer than the timeout.
+        """
+        command = self._prepare_command(
+            str(self.exe),
+            config=pathlib.Path(config),
+            requirements=pathlib.Path(requirements),
+            **kwargs,
+        )
+        try:
+            return_code = subprocess.run(command, timeout=timeout, check=True, cwd=self.exe.parent).returncode  # noqa: S603
+        except subprocess.CalledProcessError as e:
+            return_code = e.returncode
+        return (
+            return_code,
+            self._get_result(DIRECT_TEST_RESULT_FILE_NAME),
+            self._get_result(INVARIANT_TEST_RESULT_FILE_NAME),
+        )
+
+    def get_version(self) -> str:
+        """Get the version of the SDCcc executable."""
+        # use capture_output = True to get stdout and stderr instead of check_output which only collects stdout
+        process = subprocess.run([str(self.exe), '--version'], check=True, capture_output=True, cwd=self.exe.parent)  # noqa: S603
```

Comment:
line too long

Reply:
if ruff is not complaining, its not too long :D